front/app/component-library/hooks/useBreakpoint.ts

```diff
@@ -11,23 +11,23 @@ const mediaQueries = {
 type BreakpointType = keyof typeof mediaQueries;
 
 const useBreakpoint = (breakpointType: BreakpointType) => {
-  const [breakpoint, setBreakpoint] = useState(
+  const [matchesBreakpoint, setMatchesBreakpoint] = useState(
     window.matchMedia(mediaQueries[breakpointType]).matches
   );
 
   useEffect(() => {
-    const media = window.matchMedia(mediaQueries[breakpointType]);
-    if (media.matches !== breakpoint) {
-      setBreakpoint(media.matches);
-    }
+    const newMediaQueryList = window.matchMedia(mediaQueries[breakpointType]);
 
-    const listener = () => setBreakpoint(media.matches);
-    media.addEventListener('change', listener);
+    const listener = (e: MediaQueryListEvent) => {
```

Comment:
The old `listener` code worked fine too, but I thought using the event was more expressive since we're listening to an event.